presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java

```diff
@@ -171,10 +171,14 @@ public void stop()
                 .with(timeout, retry)
                 .with(executorService);
 
-        ImmutableList.copyOf(containers.values())
-                .stream()
-                .filter(DockerContainer::isRunning)
-                .forEach(container -> executor.run(container::tryStop));
+        for (DockerContainer container : ImmutableList.copyOf(containers.values())) {
```

Comment:
Why do we need to copy the list? It appears the map is not modified after this class is constructed, though the constructor should really do a defensive copy.